gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobufeditablelayer.cpp

```diff
@@ -0,0 +1,256 @@
+    auto gType = m_poFlatGeobufLayer->GetOGRGType();
+    auto createIndex = m_poFlatGeobufLayer->GetIndexNodeSize() > 0;
+    OGRFlatGeobufLayer *poFlatGeobufTmpLayer = new OGRFlatGeobufLayer(
+        osLayerName.c_str(), osTmpFilename.c_str(), spatialRef, gType, createIndex, m_papszOpenOptions);
+  
+    OGRErr eErr = OGRERR_NONE;
+    OGRFeatureDefn *poEditableFDefn = poEditableLayer->GetLayerDefn();
+    for (int i = 0; eErr == OGRERR_NONE && i < poEditableFDefn->GetFieldCount();
+         i++ )
+    {
+        OGRFieldDefn oFieldDefn(poEditableFDefn->GetFieldDefn(i));
+        int iGeomFieldIdx = 0;
+        if( (EQUAL(oFieldDefn.GetNameRef(), "WKT") &&
+             (iGeomFieldIdx = poEditableFDefn->GetGeomFieldIndex("")) >= 0) ||
+            (iGeomFieldIdx = poEditableFDefn->GetGeomFieldIndex(
+                 oFieldDefn.GetNameRef())) >= 0 )
+        {
+            OGRGeomFieldDefn oGeomFieldDefn(
+                poEditableFDefn->GetGeomFieldDefn(iGeomFieldIdx));
+            eErr = poFlatGeobufTmpLayer->CreateGeomField(&oGeomFieldDefn);
+        }
+        else
+        {
+            eErr = poFlatGeobufTmpLayer->CreateField(&oFieldDefn);
+        }
+    }
+
+    poEditableLayer->ResetReading();
+
+    // Disable all filters.
+    const char* pszQueryStringConst = poEditableLayer->GetAttrQueryString();
+    char* pszQueryStringBak = pszQueryStringConst ? CPLStrdup(pszQueryStringConst) : nullptr;
+    poEditableLayer->SetAttributeFilter(nullptr);
+
+    const int iFilterGeomIndexBak = poEditableLayer->GetGeomFieldFilter();
+    OGRGeometry* poFilterGeomBak = poEditableLayer->GetSpatialFilter();
+    if( poFilterGeomBak )
+        poFilterGeomBak = poFilterGeomBak->clone();
+    poEditableLayer->SetSpatialFilter(nullptr);
+
+    auto aoMapSrcToTargetIdx = poFlatGeobufTmpLayer->GetLayerDefn()->
+        ComputeMapForSetFrom(poEditableLayer->GetLayerDefn(), true);
+    aoMapSrcToTargetIdx.push_back(-1); // add dummy entry to be sure that .data() is valid
+
+    for( auto&& poFeature: poEditableLayer )
+    {
+        if( eErr != OGRERR_NONE )
+            break;
+        OGRFeature *poNewFeature =
+            new OGRFeature(poFlatGeobufTmpLayer->GetLayerDefn());
+        poNewFeature->SetFrom(poFeature.get(), aoMapSrcToTargetIdx.data(), true);
+        eErr = poFlatGeobufTmpLayer->CreateFeature(poNewFeature);
+        delete poNewFeature;
+    }
+    delete poFlatGeobufTmpLayer;
+
+    // Restore filters.
+    poEditableLayer->SetAttributeFilter(pszQueryStringBak);
+    CPLFree(pszQueryStringBak);
+    poEditableLayer->SetSpatialFilter(iFilterGeomIndexBak, poFilterGeomBak);
+    delete poFilterGeomBak;
+
+    if( eErr != OGRERR_NONE )
+    {
+        CPLError(CE_Failure, CPLE_AppDefined, "Error while creating %s",
+                 osTmpFilename.c_str());
+        VSIUnlink(osTmpFilename);
+        VSIUnlink(CPLResetExtension(osTmpFilename, "fgbt"));
+        return eErr;
+    }
+
+    delete m_poFlatGeobufLayer;
```

Comment:
would be safer to nullify m_poFlatGeobugLayer right away. bonus point for making m_poFlatGeobufLayer a std::unique_ptr<> and .reset()'ing it

Reply:
Tried but failed to get the bonus points.. :)